integration_k8s_kind_test/excluded_prefixes_collector_test.go

```diff
@@ -0,0 +1,213 @@
+	et.Eventually(et.checkPrefixes(expectedPrefixes), defaultTimeout, defaultTick)
+
+	expectedPrefixes = []string{"0.0.0.0/0"}
+	et.Require().NoError(exechelper.Run("kubectl replace -f ./files/updatedUserConfigMap.yaml", et.options...))
+	et.Eventually(et.checkPrefixes(expectedPrefixes), defaultTimeout, defaultTick)
+
+	et.Require().NoError(exechelper.Run("kubectl delete configmaps excluded-prefixes-config", et.options...))
+	et.Eventually(et.checkPrefixes(kubeAdmPrefixes), defaultTimeout, defaultTick)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithAllPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	et.Require().NoError(exechelper.Run("kubectl apply -f ./files/userConfigMap.yaml", et.options...))
+	defer func() {
+		et.Require().NoError(exechelper.Run("kubectl delete configmaps excluded-prefixes-config", et.options...))
+	}()
+
+	expectedPrefixes := append(append(kubeAdmPrefixes, envPrefixes...), userConfigPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), defaultTimeout, defaultTick)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithCorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	expectedPrefixes := append(kubeAdmPrefixes, envPrefixes...)
+	et.Eventually(et.checkPrefixes(expectedPrefixes), defaultTimeout, defaultTick)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithIncorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"306.306.306.0",
+	}
+
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+
+	client, err := k8s.Client()
+	et.Require().NoError(err)
+	inter := client.AppsV1().DaemonSets(collectorNamespace)
+	et.Eventually(func() bool {
+		daemonSet, err := inter.Get("excluded-prefixes-collector", metav1.GetOptions{})
+		return err == nil && daemonSet.Status.NumberAvailable == 0
+	}, defaultTimeout, defaultTick)
+}
+
+func TestExcludedPrefixesSuite(t *testing.T) {
+	suite.Run(t, &ExcludedPrefixesSuite{})
+}
+
+func (et *ExcludedPrefixesSuite) setupAlpine() {
```

Comment:
Note: Latest master has nsmgr daemonset. Please modify it instead of alpine.

Reply:
Replaced apline deployment with nsmgr daemonset